src/main.rs

```diff
@@ -7,16 +7,24 @@ extern crate rmp_serde;
 extern crate integrity_checker;
 
 use std::ffi::OsString;
+use std::io::{self, Write};
 use std::path::PathBuf;
 
-use integrity_checker::database::Database;
+use integrity_checker::database::{Database, DiffCode};
+use integrity_checker::error;
 
 enum Action {
     Build { db_path: OsString, dir_path: OsString, threads: usize },
     Check { db_path: OsString, dir_path: OsString, threads: usize },
     Diff { old_path: OsString, new_path: OsString },
 }
 
+#[derive(Debug)]
+enum ReturnCode {
+    Built,
+    Succeeded(DiffCode),
+}
```

Comment:
I was waffling on this but the more I think the more I agree with the way you did this. Really, what we're encoding here is "what happened", not just success or failure.

I'd call the enum `ActionSummary` and call the options `Built` and `Diff`.